src/border_agent/border_agent.cpp

```diff
@@ -516,4 +525,19 @@ std::string BorderAgent::GetAlternativeServiceInstanceName() const
     return ss.str();
 }
 
+#if OTBR_ENABLE_PUBLISH_MESHCOP_BR_ID
+void BorderAgent::LoadOrGenerateId(void) {
+    otError error;
+    uint16_t idLength = sizeof(mId);
+
+    error = otPlatSettingsGet(mNcp.GetInstance(), OT_SETTINGS_KEY_BR_ID, 0, mId, &idLength);
```

Comment:
I suggest we provide APIs for getting/saving this in OT instead? 
We can add APIs under `otBrAgent`?

Reply:
SGTM. @bukepo @jwhui thoughts?

Reply:
SGTM